storage/innobase/fil/fil0crypt.cc

```diff
@@ -2391,32 +2341,39 @@ static dberr_t fil_crypt_flush_space(rotate_thread_t *state) {
                           ? CRYPT_SCHEME_UNENCRYPTED
                           : crypt_data->type;
 
-  if (space->id != 0) {
-    // if (DB_SUCCESS != fil_update_encrypted_flag(space, current_type ==
-    // CRYPT_SCHEME_UNENCRYPTED ? false : true)) {
-    // ut_ad(DBUG_EVALUATE_IF("fail_encryption_flag_update_on_t3", 1, 0));
-    // return DB_ERROR;
-    //}
+  UpdateEncryptedFlagOperation update_enc_flag_op =
+      (current_type == CRYPT_SCHEME_UNENCRYPTED)
+          ? UpdateEncryptedFlagOperation::CLEAR
+          : UpdateEncryptedFlagOperation::SET;
 
-    DBUG_EXECUTE_IF("crash_on_t1_flush_after_dd_update",
-                    if (strcmp(state->space->name, "test/t1") == 0)
-                        DBUG_ABORT(););
+  if (DB_SUCCESS != fil_update_encrypted_flag(space->name, update_enc_flag_op,
```

Comment:
Can you assert that space ref_count is non_zero while we do the update operation?

Reply:
I do not see ref_count in fil_space_t, am I missing something ?